Tests/AppTests/SearchFilterTests.swift

```diff
@@ -0,0 +1,163 @@
+        }
+        
+        do { // Valid filter
+            XCTAssertTrue(parser.parse(term: "stars:5") is StarsSearchFilter)
+        }
+        
+    }
+    
+    func test_separateTermsAndFilters() {
+        let parser = SearchFilterParser()
+        let output = parser.separate(terms: ["a", "b", "invalid:true", "stars:5"])
+        
+        XCTAssertEqual(output.terms.sorted(), ["a", "b", "invalid:true"])
+        
+        XCTAssertEqual(output.filters.count, 1)
+        XCTAssertTrue(output.filters[0] is StarsSearchFilter)
+    }
+    
+    func test_binaryOperator() {
+        let matrix: [(SearchFilterComparison, Bool, SQLBinaryOperator, UInt)] = [
+            (.greaterThan, false, .greaterThan, #line),
+            (.lessThan, false, .lessThan, #line),
+            (.match, false, .equal, #line),
+            (.negativeMatch, false, .notEqual, #line),
+            
+            (.greaterThan, true, .greaterThan, #line),
+            (.lessThan, true, .lessThan, #line),
+            (.match, true, .in, #line),
+            (.negativeMatch, true, .notIn, #line),
+        ]
+        
+        matrix.forEach { comparison, isSet, sqlOperator, line in
+            XCTAssertEqual(
+                comparison.binaryOperator(isSet: isSet),
+                sqlOperator,
+                line: line
+            )
+        }
+    }
+    
+    // MARK: Filters
+    
+    func test_starsFilter() throws {
+        XCTAssertEqual(StarsSearchFilter.key, "stars")
+        XCTAssertThrowsError(try StarsSearchFilter(value: "one", comparison: .match))
+        XCTAssertEqual(try StarsSearchFilter(value: "1", comparison: .match).value, 1)
+        
+        let filter = try StarsSearchFilter(value: "1", comparison: .greaterThan)
+        let sql = renderSQL(MockPredicateBuilder.apply(filter: filter), resolveBinds: true)
+        _assertInlineSnapshot(matching: sql, as: .lines, with: """
+        "stars" > '1'
+        """)
+    }
+    
+    func test_licenseFilter() throws {
+        XCTAssertEqual(LicenseSearchFilter.key, "license")
+        XCTAssertThrowsError(try LicenseSearchFilter(value: "mit", comparison: .match))
+        XCTAssertThrowsError(try LicenseSearchFilter(value: "compatible", comparison: .greaterThan))
+        XCTAssertEqual(try LicenseSearchFilter(value: "compatible", comparison: .match).filterType, .compatible)
+        
+        let filter = try LicenseSearchFilter(value: "compatible", comparison: .match)
+        let sql = renderSQL(MockPredicateBuilder.apply(filter: filter), resolveBinds: true)
+        _assertInlineSnapshot(matching: sql, as: .lines, with: """
+        "license" IN ('afl-3.0', 'apache-2.0', 'artistic-2.0', 'bsd-2-clause', 'bsd-3-clause', 'bsd-3-clause-clear', 'bsl-1.0', 'cc', 'cc0-1.0', 'afl-3.0'0, 'afl-3.0'1, 'afl-3.0'2, 'afl-3.0'3, 'afl-3.0'4, 'afl-3.0'5, 'afl-3.0'6, 'afl-3.0'7, 'afl-3.0'8, 'afl-3.0'9, 'apache-2.0'0, 'apache-2.0'1, 'apache-2.0'2, 'apache-2.0'3, 'apache-2.0'4)
+        """)
```

Comment:
It would be fairly easy to add support for `platform:ios|macos` with the current system, it just doesn't make sense for the currently supported filters which is why it's not here.